experiments/gene-to-nissl.py

```diff
@@ -0,0 +1,169 @@
+"""Script that maps Gene Expression to Nissl."""
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import pathlib
+import sys
+from pathlib import Path
+
+import numpy as np
+from skimage.color import rgb2gray
+from utils import get_results_dir
+
+from atlannot import load_volume
+from atlannot.ants import register, transform
+
+# Initialize the logger
+logger = logging.getLogger("gene-to-nissl")
+DATA_FOLDER = pathlib.Path(__file__).resolve().parent.parent / "data"
+
+
+def parse_args():
+    """Parse arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--gene_path",
+        type=Path,
+        help="""\
+        Path to Gene Expression.
+        """,
+    )
+    parser.add_argument(
+        "--section_numbers",
+        type=Path,
+        help="""\
+        Path to json containing section numbers of gene expression.
+        """,
+    )
+    parser.add_argument(
+        "--nissl_path",
+        type=Path,
+        default=DATA_FOLDER / "ara_nissl_25.nrrd",
+        help="""\
+        Path to Nissl Volume.
+        """,
+    )
+    return parser.parse_args()
+
+
+def check_and_load(path: pathlib.Path | str) -> np.array:
+    """Load volume if path exists.
+
+    Parameters
+    ----------
+    path
+        File path.
+
+    Returns
+    -------
+    volume : np.array
+        Loaded volume.
+    """
+    if not path.exists():
+        raise ValueError(f"The specified path {path} does not exist.")
+
+    volume = load_volume(path, normalize=False)
+    return volume.astype(np.float32)
+
+
+def registration(
+    nissl_volume: np.array, gene_volume: np.array, section_numbers: np.array
+) -> np.array:
+    """Compute registration transform between a couple of volumes.
+
+    Parameters
+    ----------
+    nissl_volume
+        Nissl volume (fixed volume during registration).
+    gene_volume
+        Gene to register.(moving volume during registration).
+    section_numbers
+        Section numbers of every gene slice in gene volume.
+
+    Returns
+    -------
+    warped_genes : np.array
+        Warped slice.
+    """
+    rgb = False
+    if gene_volume.ndim == 4:
+        rgb = True
+
+    warped_genes = []
+    for i, (section_number, gene_slice) in enumerate(zip(section_numbers, gene_volume)):
+        try:
+            nissl_slice = nissl_volume[int(section_number)]
+        except IndexError:
+            continue
+
+        if rgb:
+            gene_slice_rgb = gene_slice.copy()
+            gene_slice = rgb2gray(gene_slice)
```

Comment:
Do the images have white background (as the raw images) or do we assume we flipped the colors (making it more similar to the nissl volume).

Not sure if this would influence the performance of `antspy`.